src/util/ast.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Node as AstNodeInternal,
+  JSONPath,
+  findNodeAtLocation as findNodeAtLocationInternal,
+  parseTree as parseTreeInternal,
+  ParseError,
+} from 'jsonc-parser';
+import {PlaceholderConfig} from '../analyzer.js';
+import {WireitError} from '../error.js';
+export {ParseError} from 'jsonc-parser';
+
+type ValueTypes = string | number | boolean | null | undefined;
+
+/**
+ * A JSON AST node.
+ *
+ * A safer override, preferring unknown over any.
+ */
+export interface AstNode<T extends ValueTypes = ValueTypes>
+  extends AstNodeInternal {
+  value: T;
+  children?: AstNode[];
+  parent?: AstNode<undefined>;
+}
+
+export interface ArrayNode<T> {
```

Comment:
Wondering if `AstNode` could accept `ValueTypes | ValueTypes[]` instead of there being a separate interface for `ArrayNode`? (I'm sure there's a reason, though).

Reply:
Added a comment